File: 1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#


# @lc code=start
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        pair: list[int] = []
        diff: dict[int, int] = {}
        for i, num in enumerate(nums):
            if target - num in diff:
                pair = [diff[target - num], i]
                break
            else:
                diff[num] = i
        return pair


# @lc code=end

File: 1.两数之和.ts
/*
 * @lc app=leetcode.cn id=1 lang=typescript
 *
 * [1] 两数之和
 */

// @lc code=start
function twoSum(nums: number[], target: number): number[] {
    let pair: number[] = [];
    const diff: { [n: number]: number } = {}
    for (let i = 0; i < nums.length; i += 1) {
        if (target - nums[i] in diff) {
            pair = [diff[target - nums[i]], i];
            break;
        } else {
            diff[nums[i]] = i;
        }
    }
    return pair;
};
// @lc code=end

